: constant fold None

ThomasJannaud opened the pull request:
Summary:
Constant folding should fold 'None' and consider it a constant

This goes with D74350331 and D74349918 but keeping things separate

Differential Revision: D74350331

JacobSzwejbka commented on exir/passes/constant_prop_pass.py:
```diff
@@ -61,6 +61,8 @@ def is_const(
         )
     elif isinstance(arg, _PRIMITIVE_TYPES):
         return True
+    elif arg is None:
```
was the problem before that const prop would skip nodes with None in the args?

metascroy replied:
yes, I think the issue is None is not a constant, and so if an op has args that include constants and None, it doesn't get constant folded.

ThomasJannaud committed:
: constant fold None (#10762)

Summary:

Constant folding should fold 'None' and consider it a constant

This goes with D74280168 and D74349918 but keeping things separate

Reviewed By: metascroy

Differential Revision: D74350331
